packages/plugin-essentials/sources/suggestUtils.ts

```diff
@@ -287,19 +284,22 @@ export async function getSuggestedDescriptors(request: Descriptor, {project, wor
   return suggested.slice(0, maxResults);
 }
 
-export async function fetchDescriptorFrom(ident: Ident, range: string, {project, cache, preserveModifier = true}: {project: Project, cache: Cache, preserveModifier?: boolean | string}) {
+export async function fetchDescriptorFrom(ident: Ident, range: string, {project, cache, workspace, preserveModifier = true}: {project: Project, cache: Cache, workspace: Workspace, preserveModifier?: boolean | string}) {
   const latestDescriptor = structUtils.makeDescriptor(ident, range);
 
   const report = new ThrowReport();
 
   const fetcher = project.configuration.makeFetcher();
   const resolver = project.configuration.makeResolver();
 
-  const resolverOptions = {checksums: project.storedChecksums, project, cache, fetcher, report, resolver};
+  const fetchOptions: FetchOptions = {project, fetcher, cache, checksums: project.storedChecksums, report, skipIntegrityCheck: true};
+  const resolveOptions: ResolveOptions = {...fetchOptions, resolver, fetchOptions};
+
+  const boundDescriptor = resolver.bindDescriptor(latestDescriptor, workspace.anchoredLocator, resolveOptions);
```

Comment:
Can you add a comment detailing why binding the descriptor is necessary (with an example of what would happen if we didn't)?

Reply:
Aaah I understand - I wondered why it was needed now when it wasn't before. It's because you turned the `cwd` parameter into `workspace`. Is that intended?

One problematic side effect of this change is that, let's say I run `yarn add ./foo` from a subdirectory of the active package, it will resolve `./foo` based on the root of the package rather than the current cwd.

Reply:
> It's because you turned the cwd parameter into workspace. Is that intended?

I removed the `cwd` parameter because it produced absolute paths and I didn't realize that `workspace.cwd` can be different from `this.context.cwd` (🤦‍♂️). I've fixed it now by prepending `./` to the relative path computed between `workspace.cwd` and `ppath.resolve(cwd, path)`.

> I wondered why it was needed now when it wasn't before

I think it's because `fileUtils.parseSpec` now requires a `parentLocator` (`params.locator`).